docs/SERVER.md

```diff
@@ -279,6 +222,328 @@ Server address: smp://d5fcsc7hhtPpexYUbI2XPxDbyU2d3WsVmROimcL90ss=:V8ONoJ6ICwnrZ
+
+- [Tails](https://tails.net/) live usb drive with [persistent and encrypted storage](https://tails.net/doc/persistent_storage/create/index.en.html)
+- Offline Linux laptop
+- Bitwarden
+
+Follow the steps to secure your CA keys:
+
+1. Login to your server via SSH.
+
+2. Display the content of the CA key:
+
+   ```sh
+   cat /etc/opt/simplex/ca.key
+   ```
+
+3. Write/store the content of the CA key in a safe place.
+
+4. Delete the CA key from the server. **Please make sure you've saved you CA key somewhere safe. Otherwise, you would lose the ability to [rotate the online certificate](#online-certificate-rotation)**:
+
+   ```sh
+   rm /etc/opt/simplex/ca.key
+   ```
+
+### Online certificate rotation
+
+Operators of smp servers **ARE ADVISED** to rotate online certificate regularly (every 3 months). In order to do this, follow the steps:
+
+1. Create relevant folders:
+
+   ```sh
+   mkdir -p $HOME/simplex/smp/config
+   ```
+
+1. Copy the configuration files from the server to the local machine (if not yet):
+
+   ```sh
+   rsync -hzasP <server_user>@<server_address>:/etc/opt/simplex/ $HOME/simplex/smp/config/
+   ```
+
+2. **Copy** your CA private key from a safe place to the local machine and name it `ca.key`.
+
+3. Download latest `smp-server` binary [from Github releases](https://github.com/simplex-chat/simplexmq/releases):
+
+   ```sh
+   curl -L 'https://github.com/simplex-chat/simplexmq/releases/latest/download/smp-server-ubuntu-20_04-x86-64' -o smp-server
+   ```
+
+4. Put the `smp-server` binary to your `$PATH` and make it executable:
+
+   ```sh
+   sudo mv smp-server /usr/local/bin/ && chmod +x /usr/local/bin/smp-server
+   ```
+
+5. Export a variable to configure your path to smp-server configuration:
+
+   ```sh
+   export SMP_SERVER_CFG_PATH=$HOME/simplex/smp/config
+   ```
+
+6. Execute the following command:
+
+   ```sh
+   smp-server cert
+   ```
+
+   This command should print:
+
+   ```sh
+   Certificate request self-signature ok
+   subject=CN = <your domain or IP>
+   Generated new server credentials
+   ----------
+   You should store CA private key securely and delete it from the server.
+   If server TLS credential is compromised this key can be used to sign a new one, keeping the same server identity and established connections.
+   CA private key location:
+   $HOME/simplex/smp/config/ca.key
+   ----------
+   ```
+
+7. Remove the CA key from the config folder (make sure you have a backup!):
+
+   ```sh
+   rm $HOME/simplex/smp/config/ca.key
+   ```
+
+8. Upload new certificates to the server:
+
+   ```sh
+   rsync -hzasP $HOME/simplex/smp/config/ <server_user>@<server_address>:/etc/opt/simplex/
```

Comment:
this may overwrite any configuration changes made on the server, better to use commands that only copy key and cert

Reply:
or maybe it's ok it was copied as a whole before...